Feature digital input selector module

nsturgeon19 opened the pull request.

nsturgeon19 committed:
Made a DI selecter module

nsturgeon19 committed:
Got incredibly confused

nsturgeon19 committed:
Added some arraylists

nsturgeon19 committed:
Added an addDigitalInput method and a currentState method

nsturgeon19 committed:
Removed a return

david-boles reviewed (changes requested):
Synchronize everything
Remove return for addDigitalInput
Remove //TODO comments
Add logging, look at anything in the core already

david-boles reviewed (changes requested):
Add testing

nsturgeon19 committed:
Removed TODO's

nsturgeon19 committed:
Updating to Develop (#62)

* Fix the docs

Really!? Misspelled "exception"!? Also, don't document things that don't
exist

* Feature: generic controls module improvements (#51)

* added button inversion and remeoved hash table and added javadocs

* David wanted radians

* Added Axes inversion 
David wanted radians

* ModuleHashTable: Fixed getAllSubModules return type.

* Fixed some things

* added more functionallity (#55)

* added more functionallity

* added suggested changes

* win_build: fixed

* Made edits
David wants radians

* Should be functional, need david to check and should test, but otherwise will work. Needs javadocs, but will do that on eclipse.
David still wanted radians.

* Added Javadocs
David wants radians even though he wants to save memory!

* ModuleHashTable: Fixed location of initModule javadoc in constructor.

* Fixed Enums

* Feature fix docs (#57)

* Remove doc for nonexistent param

* Remove errored @return

* Fix malformed html?

* Fix doc errors

* Fix param errors

* Fix nsturgeon's crappy spelling

* Make the gitignore work better (#14)

* added functionality for roborio accelerometer (#58)

* added functionality for roborio accelerometer

* added axis swapping and inversion

* added more comments.

* Refactored shifters into package, improved Double...

* Feature: Logging overhaul and implementations (#61)

* LoggerModule: Added methods for verbose logging.

* GenericControlsModule: Added verbose logging support.

* SimpleProcedureModule: Added verbose logging.

* SimpleProcedureModule: Fixed constructor logger source.

* SimpleModule: Added support for verbose logging.

* SimpleModule: Added constructor arguments to creation log.

* LoggerModule: Removed IllegalArgumentExceptions

* RobotModule: Added verbose logging support.

* SimpleOwnableModule: Added verbose logging support.

* DoubleSolenoidShifter: Added support for verbose logging.

* Changed init and shutD logging.

* Changed init and shutD logging.

* RoborioAccelerometerModule: Added verbose logging support.

* LoggerModule: Added methods and organized.

* Created NullLoggerModule

* GenericModule: Added verbosity control.

* Created SystemLoggerModule

* Added verbosity control to LoggerModule

nsturgeon19 committed:
Added loggers

nsturgeon19 committed:
Added javadocs

nsturgeon19 committed:
Added a prompAndGetReturn and a PromptAndGetReturnBoolean

nsturgeon19 committed:
Added testing code

nsturgeon19 committed:
Added Javadocs

nsturgeon19 committed:
Added a logger in the consturcter
